sky/utils/controller_utils.py

```diff
@@ -805,6 +805,16 @@ def _sub_path_join(sub_path: Optional[str], path: str) -> str:
     else:
         (store_type, bucket_name, sub_path, storage_account_name, region) = (
             storage_lib.StoreType.get_fields_from_store_url(bucket_wth_prefix))
+        cloud_str = store_type.to_cloud()
+        if (cloud_str not in
+                storage_lib.get_cached_enabled_storage_clouds_or_refresh()):
```

Comment:
Not sure if it would cause problems for S3-compatible storage like MinIO

Reply:
It's ok for now, since we currently consider s3-compatible storage to be it's own cloud, e.g., cloudflare r2. We should do the refactoring in the future.